File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>


int main(void)
{
    string text = get_string("Text: ");

    int length = strlen(text);
//count letters ***************************************
    int letters = 0;
    for (int i = 0; i <= length; i++)
    {
        if (isalnum(text[i]))
        {
            letters++;
        }
        else
        {
            continue;
        }
    }

//count words ***************************************
    int words = 0;
    for (int i = 0; i <= length; i++)
    {
        if ((text[i]) == ' ' && (text[i + 1] != ' '))
        {
            words++;
        }
        else
        {
            continue;
        }
    }

//count sentences ***************************************
    int sentences = 0;
    for (int i = 0; i <= length; i++)
    {
        if ((text[i]) == '.' || (text[i] == '!') || (text[i] == '?'))
        {
            sentences++;
        }
        else
        {
            continue;
        }
    }

//summary ***************************************
    // printf("%d letters\n", letters);
    // printf("%d words\n", words+1);
    // printf("%d sentences\n", sentences);
    float L = (float)letters / (words + 1) * 100;
    float S = (float)sentences / (words + 1) * 100;
    float beforeindex = (0.0588 * L) - (0.296 * S) - 15.8;
    int index = round(beforeindex);
    // printf("%.2f\n", L);
    // printf("%.2f\n", S);
    // printf("%.2f\n", beforeindex);
    // printf("%d\n", index);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }
}

File: sentimental-readability/readability.py
# TODO

from cs50 import get_string

string = get_string("Text: ")
# def
letter = 0
word = 0
sentence = 0
# check letters
for c in string:
    if c == '\n':
        exit()
    if c.lower().isalpha() == True:
        letter += 1
    if c == " ":
        word += 1
    if c == "." or c == "!" or c == "?":
        sentence += 1
# calculate Coleman-Liau index
word += 1
L = (letter/word)*100
S = (sentence/word)*100

grade = round(0.0588 * L - 0.296 * S - 15.8)

# print result
if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print("Grade", grade)